src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt

```diff
@@ -171,8 +173,8 @@ open class CsmRbac(
       rolesDefinition: RolesDefinition
   ): Boolean {
     logger.debug("RBAC ${rbacSecurity.id} - Verifying default roles for permission: $permission")
-    val isAuthorized =
-        this.verifyPermissionFromRole(permission, rbacSecurity.default, rolesDefinition)
+    val defaultRole = if (rbacSecurity.default == ROLE_NONE) null else rbacSecurity.default
```

Comment:
I don't think that you need this line :thinking: 
If you leave `ROLE_NONE` as `defaultRole` value:
- no need to consider that role is nullable thought all verify* method calls
- this.getRolePermissions("ROLE_NONE", rolesDefinition.permissions) will returned an empty list
- verifyPermission(permission: String, userPermissions: List<String>) will return false if userPermissions is an empty list (which is the expected behavior)

Reply:
Discussed & resolved by call